### PR TITLE
build(docs-infra): several improvements to `NgPackagesInstaller`

### PR DESCRIPTION
**Pull request by gkalpak**
Several improvements to `NgPackagesInstaller` (see individual commit messages).
Mainly, making it more predictable wrt to always installing the latest locally built Angular packages.

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]
No fine tooth comb review - sorry.
But a 5 min read-through looks good to me.

File: aio/tools/ng-packages-installer/index.js [hunk 1]
```diff
@@ -160,6 +162,30 @@ class NgPackagesInstaller {
...
+    const canBuild = process.platform !== 'win32';
+
+    if (canBuild) {
+      this._log(`Building the Angular packages with: ${ANGULAR_DIST_PACKAGES_BUILD_CMD}`);
+      shelljs.exec(ANGULAR_DIST_PACKAGES_BUILD_CMD);
+    } else {
+      this._warn([
+        'Building the local Angular packages is currently not supported on Windows.',
+        `Please, ensure '${ANGULAR_DIST_PACKAGES}' exists and is up-to-date (e.g. by running ` +
+          `'${ANGULAR_DIST_PACKAGES_BUILD_CMD}' in Windows Subsystem for Linux or a Linux docker container or VM).`,
```
Comment (devversion) [inline aio/tools/ng-packages-installer/index.js]: FWIW: I'm running Bazel on Windows w/ Git Bash and it works to run that script. Maybe we should expand the warning message to say that the package installer is not able to build the packages automatically?

I realize that it's difficult to run the script automatically, but technically bash is a prerequisite for working on Angular right now (as Bazel also requires bash on windows through the `BAZEL_SH` environment variable)

Comment (gkalpak) [reply aio/tools/ng-packages-installer/index.js]: The truth is that the build script does not currently support windows (and running through a thousand hoops to make it work isn't "support" 😛).

I am happy to mention `Git bash` as an option, if you say it works there (I'll take your word on this one :grin:).

Note that, while `bash` and `bazel` are prerequisites for working on Angular's source code, they are _not_ (currently) required for `aio/`, which is being worked on by people on Windows (e.g. docs authors) much more than the rest of the codebase :wink:

Comment (gkalpak) [reply aio/tools/ng-packages-installer/index.js]: Added `Git Bash for Windows` as a possible option and changed the warning to say _"**Automatically** building [...] is currently not supported_:

```diff
-Building the local Angular packages is currently not supported on Windows.
+Automatically building the local Angular packages is currently not supported on Windows.
```

WDYT?

Comment (devversion) [reply aio/tools/ng-packages-installer/index.js]: Yeah that's similiar to what I had in mind. Thanks!

I think we should be able to convert the Bash script eventually to a NodeJS script. Might be something to explore 😆.

**Comment by alxhub** [comment]
@gkalpak this fails to merge on the patch branch. Can you open a separate PR for it? I'll proceed and merge this to master.

**Comment by gkalpak** [comment]
Sure, there you go: #32013

**Pull request closed without merging**